### DISCUSSION TITLE
Is "type" keyword accepted as identifier ?

### DISCUSSION THREAD

**Discussion by mingodad** [teal-language/tl#511] [Q&A] [answered]
Trying to write an alternative parser for `tl` I found a possible bug on https://github.com/teal-language/tl/blob/ef4fcda5ba0a0c83d2a4dbfac67baf186e8c4252/tl.tl#L46 we have as field of a record but on https://github.com/teal-language/tl/blob/ef4fcda5ba0a0c83d2a4dbfac67baf186e8c4252/tl.tl#L54 and https://github.com/teal-language/tl/blob/ef4fcda5ba0a0c83d2a4dbfac67baf186e8c4252/tl.tl#L1201 we have `type` as field name for another record.

This makes difficult to parse because `type` seems to be a keyword that also is valid inside a record for type declaration like here https://github.com/teal-language/tl/blob/ef4fcda5ba0a0c83d2a4dbfac67baf186e8c4252/tl.tl#L10 inside `local record tl`.

Is this intended or a bug ?

**Comment by hishamhm** [comment, marked as answer]
Yes, it is intended and not a bug. `type` is not a hard keyword, it is a [soft keyword](https://stackoverflow.com/questions/65800344/what-are-soft-keywords) (i.e., context-sensitive). All Teal keyword additions are intended as soft keywords, to minimize compatibility issues with existing Lua libraries.

**Reply by mingodad** [reply]
Thanks for clarify it !
If so why not be consistent and change here https://github.com/teal-language/tl/blob/ef4fcda5ba0a0c83d2a4dbfac67baf186e8c4252/tl.tl#L46

**Reply by hishamhm** [reply]
There's no need to, and sometimes the "inconsistency" makes it easier for me to spot which table is which. Also, I use `typ` in lots of places as locals (where I can't use `type`) and it's nicer to match these, but `Result` is more of a "public" interface.

Ultimately it's a matter of taste, of course, but I'm happy with the code as is.

**Reply by mingodad** [reply]
Thanks for reply !
I've got a kind of alternative parser for `Teal` and with it I did a crude initial conversion to what could be the `LJS` (https://github.com/mingodad/ljs) version of it and anyone interested can see it here https://gist.github.com/mingodad/1a5b0ed6c25232cb29db9e3fa7de61b1 , I also successfully converted `tl.lua` to `LJS` with the `lua2ljs` tool (https://github.com/mingodad/ljs/tree/master/lua2ljs) , I'm not sure if I will go forward with making a `Tealjs` version of `Teal`.
